app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -6,23 +6,67 @@
 
 package mozilla.lockbox.presenter
 
+import android.widget.EditText
+import io.reactivex.Observable
 import io.reactivex.rxkotlin.addTo
+import mozilla.lockbox.R
 import mozilla.lockbox.flux.Dispatcher
 import mozilla.lockbox.flux.Presenter
 import mozilla.lockbox.model.ItemDetailViewModel
 import mozilla.lockbox.model.titleFromHostname
 import mozilla.lockbox.store.DataStore
+import android.content.ClipData
+import android.content.ClipboardManager
 
 interface ItemDetailView {
     var itemId: String?
     fun updateItem(item: ItemDetailViewModel)
+    fun copyNotification(strId: Int)
+    fun updatePasswordField(visible: Boolean)
+
+    val btnUsernameCopyClicks: Observable<Unit>
+    val editUsername: EditText
+
+    val btnPasswordCopyClicks: Observable<Unit>
+    val editPassword: EditText
+
+    val btnTogglePasswordClicks: Observable<Unit>
 }
 
 class ItemDetailPresenter(
     private val view: ItemDetailView,
+    private val clipboardManager: ClipboardManager,
     private val dispatcher: Dispatcher = Dispatcher.shared,
     private val dataStore: DataStore = DataStore.shared
+
 ) : Presenter() {
+
+    var isPasswordVisible: Boolean = false
+
+    override fun onViewReady() {
+
+        this.view.btnUsernameCopyClicks
+                .subscribe {
+                    clipboardCopy("username", view.editUsername.text.toString())
+                    view.copyNotification(R.string.toast_username_copied)
+                }
+                .addTo(compositeDisposable)
+
+        this.view.btnPasswordCopyClicks
+                .subscribe {
+                    clipboardCopy("password", view.editPassword.text.toString())
```

Comment:
I think we should be using the `ServerPassword` or the `ItemDetailViewModel` version of the username/password, rather than the UI version.